src/chocolatey/infrastructure.app/nuget/NugetPush.cs

```diff
@@ -40,7 +41,9 @@ public static void push_package(ChocolateyConfiguration config, string nupkgFile
             const bool disableBuffering = false;
 
             // Controls adding /api/v2/packages to the end of the source url
-            const bool noServiceEndpoint = true;
+            const bool noServiceEndpoint = false;

```

Comment:
We may need to do some further parsing/checking here, assuming this change fixes pushing to CCR, since I think it break would break pushing to Nexus (IIRC).

Reply:
That's what I'm not entirely sure on. Right now I'm not able to push to a v3 feed from Visual Studio, it prompts me for username/password no matter what (even reverting to the head of develop). But from my tests against v2 it had no problem. I'm running this on Test Kitchen right now to see how it behaves with pushing to the v3 feeds there.

Assuming that works, great, if not then I'm thinking we just wrap this:

```csharp
            // Controls adding /api/v2/packages to the end of the source url
            bool noServiceEndpoint = true;

            if (config.Sources == ApplicationParameters.ChocolateyCommunityFeedPushSource
                || config.Sources == ApplicationParameters.ChocolateyCommunityFeedPushSourceOld)
            {
                noServiceEndpoint = false;
            }
```

Reply:
Actually, I can explain why it being set to false is sufficient. When it's building the URI to push to, it looks at the absolute path, and if the absolute path after removing the starting `/` characters is empty, and `noServiceEndpoint` is `false`, then it sets the path to `/api/v2/package/` plus whatever it was asked to append (in our case nothing). So when you already have it going to say `https://nexus.local.testing/repository/test/` then the absolute path is `/repository/test/`. So it is only expected to add `/api/v2/package/` if you're going to a direct hostname (`https://push.chocolatey.org`, `https://nuget.org`, etc)

The relevant line: https://github.com/chocolatey/NuGet.Client/blob/40e0be43ff5afc6962fac29156942c4e9650a979/src/NuGet.Core/NuGet.Protocol/Resources/PackageUpdateResource.cs#L808

Reply:
Based on this comment, and additional conversations around this topic, going to move forward with getting this merged.